Client/Frontend/Sync/BraveCore/BraveCoreMigrator.swift

```diff
@@ -12,14 +12,17 @@ import CoreData
 
 private let log = Logger.browserLogger
 
+// MARK: MigrationSyncTypes
+
+public enum MigrationSyncTypes {
+    case bookmarks
+    case history
+    case all
+}
```

Comment:
Thats a great observation, thanks for checking it deeply. I did it like this way because I would like to add async migration between different types so one type will not wait for the other. Actually I have the code implemented but I had problem with DCHECK On brave-core side.

Changes made as you suggested which includes the CoreMigration method using interstitialspage.
Fix at https://github.com/brave/brave-ios/pull/3703/commits/0c734ac8b97c639db3100e17e1a12a675caf7433